packages/orm/src/EntityManager.ts

```diff
@@ -520,17 +521,21 @@ export class EntityManager {
   }
 
   /** Registers a newly-instantiated entity with our EntityManager; only called by entity constructors. */
-  register(entity: Entity): void {
-    if (entity.id && this.findExistingInstance(getMetadata(entity).cstr, entity.id) !== undefined) {
+  register(meta: EntityMetadata<any>, entity: Entity): void {
```

Comment:
The caller already had `meta` around so passing it through to avoid a `getMetadata` call.